lstchain/reco/reconstructor.py

```diff
@@ -688,89 +686,54 @@ def array_times_template(array, ti, template, is_high_gain):
-        sigma_n = sigma_n.T
-        logger.debug("old mean %s; sigma n %s", mean, sigma_n)
-
         sigma_n = (self.error**2)[..., None] + sigma_n
         sigma_n = np.sqrt(sigma_n)
-
         log_gauss = log_gaussian(signal[..., None], mean, sigma_n)
 
         if np.any(~mask):
-            mu_hat_LG = ((mu[~mask] / (1-self.crosstalk_factor[~mask]))
-                         * (self.gain[~mask][..., None] * self.template[~mask](t, gain='LG')))
-            mu_hat_HG = ((mu[~mask] / (1-self.crosstalk_factor[~mask]))
-                         * (self.gain[~mask][..., None] * self.template[~mask](t, gain='HG')))
-            mu_hat = (mu_hat_HG.T * self.is_high_gain[~mask]) + mu_hat_LG.T * (~self.is_high_gain[~mask])
-            mu_hat = mu_hat.T
-
-            sigma_hat_LG = ((mu[~mask] / np.power(1-self.crosstalk_factor[~mask], 3))
-                            * (self.gain[~mask][..., None]
-                               * self.template[~mask](t, gain='LG')) ** 2)
-            sigma_hat_HG = ((mu[~mask] / np.power(1-self.crosstalk_factor[~mask], 3))
-                            * (self.gain[~mask][..., None]
-                               * self.template[~mask](t, gain='HG')) ** 2)
-            sigma_hat = (sigma_hat_HG.T * self.is_high_gain[~mask]) + sigma_hat_LG.T * (~self.is_high_gain[~mask])
-            sigma_hat = sigma_hat.T
-            sigma_hat = np.sqrt((self.error[~mask]**2)[..., None] + sigma_hat)
-
-            """
-                    Alternative to test (done in independent code and above)
-            mu_hat = ((mu[~mask] / (1-self.crosstalk_factor[~mask]))
-                     * array_times_template(t, self.gain[~mask], self.template[~mask],
-                                     self.is_high_gain[~mask])[..., None])
-            sigma_hat = (((mu[~mask] / np.power(1-self.crosstalk_factor[~mask], 3))
-                         * array_times_template(t, self.gain[~mask], self.template[~mask],
-                                         self.is_high_gain[~mask])**2)[..., None])
-            """
-
-            log_pixel_pdf_HL = log_gaussian(signal[~mask][..., None], mu_hat, sigma_hat)
+            mu_hat = ((mu[~mask] / (1-self.crosstalk[~mask]))[..., None]
+                     * array_times_template(self.gain[~mask], t[~mask], self.template,
+                                     self.is_high_gain[~mask]))
+            sigma_hat = (((mu[~mask] / np.power(1-self.crosstalk[~mask], 3))[..., None]
+                         * array_times_template(self.gain[~mask], t[~mask], self.template,
+                                         self.is_high_gain[~mask])**2))
+            sigma_hat = np.sqrt((self.error[~mask]**2) + sigma_hat)
+
+            log_pixel_pdf_HL = log_gaussian(signal[~mask], mu_hat, sigma_hat)
 
         log_poisson = np.expand_dims(log_poisson.T, axis=1)
         log_pixel_pdf_elt = log_poisson + log_gauss
         pixel_pdf = np.sum(np.exp(log_pixel_pdf_elt), axis=-1)
 
         log_pdf2 = 0
+        a=False
```

Comment:
this is temporary plug for testing only, right?

Reply:
yes